### ISSUE TITLE
Segmentation fault on `get_pixels` on Ruby-2.2

### ISSUE THREAD

**Issue by ShockwaveNN** [rmagick/rmagick#212]
1. Create `test.rb` this code
   
   ```
   require 'rmagick'
   include Magick
   
   (0..10).each do |_|
     ImageList.new('1.bmp').get_pixels(0, 0, 1600, 800).each_slice(1600).to_a
   end
   ```
2. Create any `1.bmp` file with dimenstion 1600*800
3. Install ruby 2.1.5 (via RVM for example) and run 

```
  ruby -v # ruby 2.1.6p336 (2015-04-13 revision 50298) [x86_64-linux]
  ruby test.rb # Nothing output, all good
```
1. Install ruby 2.2.2 (via RVM for example) and run

```
  ruby -v #ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-linux]
  ruby test.rb # You get a big log of Segmentation failed
```

I'm not sure this is bug in rmagick specifically, but not in ruby itself.

My config is Kubunut 15.04 with:

```
gem list | grep rmagick #rmagick (2.15.3)
dpkg -l | grep imagema
ii  imagemagick                                                 8:6.8.9.9-5                                amd64        image manipulation programs -- binaries
ii  imagemagick-6.q16                                           8:6.8.9.9-5                                amd64        image manipulation programs -- quantum depth Q16
ii  imagemagick-common                                          8:6.8.9.9-5                                all          image manipulation programs -- infrastructure

```

**Comment by ShockwaveNN** [comment]
Problem still actual with `ruby-2.2.3` and `rmagick (2.15.4)`

**Comment by Zechtitus** [comment]
Seeing the same thing with 2.3 and rmagick 2.15.4
`/var/lib/gems/2.3.0/gems/rmagick-2.15.4/lib/rmagick_internal.rb:832: [BUG] Segmentation fault at 0x0000000000001c
ruby 2.3.0p0 (2015-12-25) [x86_64-linux-gnu]`

...

`/var/lib/gems/2.3.0/gems/rmagick-2.15.4/lib/rmagick_internal.rb:832:in `each_pixel'
/var/lib/gems/2.3.0/gems/rmagick-2.15.4/lib/rmagick_internal.rb:832:in `get_pixels'
`

**Comment by ShockwaveNN** [comment]
Fixed in Ruby 2.3.5 and Ruby 2.4.2
See: https://bugs.ruby-lang.org/issues/12670
